File: New version/VHKPlayer/VHKPlayer/Commands/Logic/CommandProcessor.cs
using Autofac;
using VHKPlayer.Commands.Logic.Interfaces;

namespace VHKPlayer.Commands.Logic
{
    public class CommandProcessor : ICommandProcessor
    {
        private readonly IComponentContext context;

        public CommandProcessor(IComponentContext context)
        {
            this.context = context;
        }

        public void Process(ICommand command)
        {
            var handlerType = typeof(ICommandHandler<>).MakeGenericType(command.GetType());
            dynamic handler = context.Resolve(handlerType);

            handler.Handle((dynamic)command);
        }
    }
}

File: core/src/main/java/com/vhkplayer/dao/commands/CommandProcessor.java
package com.vhkplayer.dao.commands;

import java.util.Map;

public final class CommandProcessor {
    private Map<Class, CommandHandlerCreationStrategy> handlerOptions;

    public CommandProcessor(Map<Class, CommandHandlerCreationStrategy> handlerOptions) {
        this.handlerOptions = handlerOptions;
    }

    public <TCommand extends Command> void process(TCommand command) {
        CommandHandlerCreationStrategy<TCommand> creationStrategy = handlerOptions.get(command.getClass());
        if (creationStrategy != null) {
            creationStrategy.createHandler().handle(command);
        }
    }
}
